cloud-aws/src/main/resources/templates/aws-cf-stack.ftl

```diff
@@ -533,6 +533,28 @@
         </#if>
         "ThroughputMode": "${efsFileSystem.throughputMode}"
       }
+    },
+    "MountTarget": {
+      "Type": "AWS::EFS::MountTarget",
+      "Properties": {
+        "FileSystemId": { "Ref": "ElasticFileSystem" },
+        <#if !existingSubnet>
+            "SubnetId": { "Ref": "PublicSubnet" },
+        <#else>
+            "SubnetId": { "Ref": "SubnetId" },
+        </#if>
+        "SecurityGroups" : [
+          <#list instanceGroups as group>
+            <#if group.cloudSecurityIds?size != 0>
```

Comment:
I am not sure if it is a valid scenario, but imagine that some instancegroup has prescribed SGs (say `sg-1` and `sg-2`), while some other instancegroup does not have a predefined SG (and thus gets an SG created above like `ClusterNodeSecurityGroupNotExisting`). Then this logic will generate a JSON fragment as below:

```
"SecurityGroups" : [
"sg-1",
"sg-2",
{ "Ref" : "ClusterNodeSecurityGroupNotExisting" }
]
```
So the list will be heterogeneous, some items being simple strings, while others being objects (with a single `"Ref"` property). If this is indeed a valid scenario, I wonder if CF accepts such a JSON.

Reply:
@doktoric, @topolyai5   Is it possible to encounter the scenario described above?

Reply:
After talking to Richard, it seems likely that custom DLs do permit such situations. So even it does not affect customers, this internal scenario still needs to be covered.

Reply:
I have confirmed that CloudFormation does NOT allow more than 5 security groups associated with EFS.
I have an instance group with 5 existing security groups, and another instance group with newly created security group. they will all be associated with the mount target of the EFS. And the provision of data lake failed.

"DATA LAKE STATUS
 Provisioning Failed
REASON
Data Lake creation failed: CloudFormation stack lina-0215-2-dl-184 creation failed. Resource never entered the desired state as it failed. The following resource(s) failed to create: [AmbariNodesmaster, ElasticFileSystem, ClusterNodeSecurityGroupidbroker]. AmbariNodesmaster: The security group 'sg-02fe9cf52ed86b28b' does not exist (Service: AmazonAutoScaling; Status Code: 400; Error Code: ValidationError; Request ID: 4990c62f-c269-483b-abee-00e3778d192e; Proxy: null), ClusterNodeSecurityGroupidbroker: Resource creation cancelled, ElasticFileSystem: Resource creation cancelled"

From AWS console, EFS was created, but it has no mount target because a mount target does not allow more than 5 security groups.

Reply:
![EFS mount target not created due to more than 5 security groups](https://user-images.githubusercontent.com/26745892/108001294-a00c7e80-6fb1-11eb-9d99-e86cd851964e.png)

Reply:
Thanks for checking! 👍